core/src/main/java/org/mule/runtime/core/privileged/routing/RoutingResult.java

```diff
@@ -42,13 +46,17 @@ public RoutingResult(Map<String, Message> successfulRoutesResultMap, Map<String,
   }
 
   public static RoutingResult routingResultWithException(Map<String, Message> successfulRoutesResultMap,
-                                                         Map<String, Pair<Error, EventProcessingException>> failedRoutesErrorWithExceptionMap) {
+                                                         Map<String, Pair<Error, EventProcessingException>> errorMap) {
     RoutingResult routingResult = new RoutingResult(successfulRoutesResultMap, emptyMap());
+    Map<String, Pair<Error, MuleException>> failedRoutesErrorWithExceptionMap = errorMap.entrySet().stream()
+        .collect(Collectors
+            .toMap(Entry::getKey,
+                   pair -> new Pair<>(pair.getValue().getFirst(), getRootMuleException(pair.getValue().getSecond()))));
```

Comment:
Why are you adding getRootMuleException here?

Reply:
See previous comment.